mesa/mesaPROTON_OC.py

```diff
@@ -426,6 +432,147 @@ def setup_siblings(self):
+        self.population = self.schedule.agents
+        self.hh_size = self.household_sizes(self.initial_agents)
+        self.complex_hh_sizes = list()
+        max_attempts_by_size = 50
+
+        for size in self.hh_size:
+            success = False
+            nb_attempts = 0
+            while not success and nb_attempts < max_attempts_by_size:
+                hh_members = list()
+                nb_attempts += 1
+                head_age = \
+                extra.pick_from_pair_list(head_age_dist[head_age_dist["size"] == size][["age", "p"]].values.tolist(),
+                                          self.rng)[0]
+                if size == 1:
+                    male_wanted = (self.rng.random() < proportion_of_male_singles_by_age[
+                        proportion_of_male_singles_by_age["age"] == head_age]["p_male"].values)[0]
+                    head = self.pick_from_population_pool_by_age_and_gender(head_age, male_wanted)
+                    if head:
+                        success = True
+                else:
+                    hh_type = extra.pick_from_pair_list(
+                        hh_type_dist[hh_type_dist["age"] == head_age][["type", "p"]].values.tolist(), self.rng)[0]
+                    if hh_type == "single_parent":
+                        male_head = self.rng.random() < float(p_single_father.columns.to_list()[0])
+                    else:
+                        male_head = True
+
+                    if male_head:
+                        mother_age = extra.pick_from_pair_list(
+                            partner_age_dist[partner_age_dist["age_of_head"] == head_age][
+                                ["age_of_partner", "p"]].values.tolist(), self.rng)[0]
+                    else:
+                        mother_age = head_age
+                    hh_members.append(self.pick_from_population_pool_by_age_and_gender(head_age, male_head))
+                    if hh_type == "couple":
+                        mother = self.pick_from_population_pool_by_age_and_gender(mother_age, False)
+                        hh_members.append(mother)
+                    num_children = size - len(hh_members)
+                    for child in range(1, int(num_children) + 1):
+                        sub_num_child = children_age_dist[children_age_dist["child_number"] == num_children]
+                        if num_children in sub_num_child["child_number"] and mother_age in sub_num_child[
+                            "age_of_mother"]:
+                            child_age = extra.pick_from_pair_list(
+                                sub_num_child[sub_num_child["age_of_mother"] == mother_age][
+                                    ["age_of_child", "p"]].values.tolist(), self.rng)[0]
+                            child = self.pick_from_population_pool_by_age(child_age)
+                            hh_members.append(child)
+                        else:
+                            hh_members.append(None)
```

Comment:
We can already break the look here - the need to calculate it all then check for nobodies was caused by the functional programming style (which is a good thing), here we're procedural so we can just stop trying.